src/Engine/Scene/Parsing/SceneLoader.cpp

```
@@ -604,23 +610,107 @@ void SceneLoader::processScene(const YAML::Node &rootNode, std::string &currentE
+            // Deserialize
+                // Set up diagnostics in case of error
+            auto mark = ctx.selfComponents->at(ctx.currentComponentType).Mark();
+            int line = -1, column = -1;
+            std::string errLocation;
+            if (!mark.is_null()) {
+                line = mark.line + 1;
+                column = mark.column + 1;
+                errLocation = "At line " + std::to_string(line) + ", column " + std::to_string(column) + ": \n\n";
+            }
+            else
+                while (m_errorMarkers.contains(line))
```

Cppcheck (2.20.0)

[style] The function 'on_bright_red' is never used.

(unusedFunction)